### PR TITLE
Fixes #9971 - improve handling of id search errors

### PR DESCRIPTION
**Pull request by mbacovsky**
Do not hide search failures when intentionaly looking for non-required resource id

### CODE REVIEW

**Comment by mbacovsky** [comment]
Successful BATs tests run: http://ci.theforeman.org/job/systest_foreman_hammer_nightly/106/

File: lib/hammer_cli_foreman/commands.rb [hunk 1]
```diff
@@ -157,7 +160,10 @@ def get_resource_ids(resource, options={})
       begin
         resolver.send("#{resource.singular_name}_ids", opts)
       rescue HammerCLIForeman::MissingSeachOptions => e
-        raise e unless (options[:required] == false)
+        if (options[:required] == true || resource_search_requested(resource, opts, plural=true))
```
Comment (tstrachota) [inline lib/hammer_cli_foreman/commands.rb]: Please replace `plural=true` with 'true' only. It behaves a bit differently from what I think you expect.

**Comment by tstrachota** [comment]
LGTM, one note in the code. Other than that :+1:

**Comment by mbacovsky** [comment]
Concerned code fixed, thanks for the review, @tstrachota

**Pull request merged**